miden-tx/src/host/note_tracker.rs

```diff
@@ -0,0 +1,41 @@
+use alloc::vec::Vec;
+
+use miden_objects::{assets::Asset, notes::NoteMetadata, Digest, Felt};
+
+// OUTPUT NOTE BUILDER
+// ================================================================================================
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct OutputNoteData {
```

Comment:
Not sure if this is the most elegant solution. Maybe a normal dict would have sufficed?

Reply:
What do you mean? Why do you want to use a map here?